specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/preview/2021-07-01-preview/nodetype.json

```diff
@@ -0,0 +1,1396 @@
+            "$ref": "./examples/ReimageNodes_example.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "responses": {
+          "202": {
+            "description": "The request was accepted and the operation will complete asynchronously."
+          },
+          "200": {
+            "description": "The operation completed successfully."
+          },
+          "default": {
+            "description": "The detailed error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorModel"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabric/managedClusters/{clusterName}/nodeTypes/{nodeTypeName}/deleteNode": {
+      "post": {
+        "tags": [
+          "NodeType"
+        ],
+        "operationId": "NodeTypes_DeleteNode",
+        "summary": "Deletes one or more nodes on the node type.",
+        "description": "Deletes one or more nodes on the node type. It will disable the fabric nodes, trigger a delete on the VMs and removes the state from the cluster.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/clusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/nodeTypeNameParameter"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "parameters for delete action.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/NodeTypeActionParameters"
+            },
+            "x-ms-parameter-location": "method"
+          }
+        ],
+        "x-ms-examples": {
+          "Delete nodes": {
+            "$ref": "./examples/DeleteNodes_example.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "responses": {
+          "202": {
+            "description": "The request was accepted and the operation will complete asynchronously."
+          },
+          "200": {
+            "description": "The operation completed successfully."
+          },
+          "default": {
+            "description": "The detailed error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorModel"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabric/managedClusters/{clusterName}/nodeTypes/{nodeTypeName}/skus": {
+      "get": {
```

Comment:
>"get" [](http://example.com/codeflow?start=6&length=5)

Please consider using POST action to return metadata like skus. e.g. /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabric/managedClusters/{clusterName}/nodeTypes/{nodeTypeName}/getSkus

If you model this as a child resource type, you will need to support GET on a single sku as well. Also it needs to return top level properties like type, id, name of a resource.

[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

Reply:
I don't see anywhere on the link you shared where it suggests using Post for this case. Could you point me to the exact place or explain further? This is not a tracked child resource, Are we still required to return type, id, name, etc?

Reply:
Here is a RPC about actions. Generally if the api does not return manageable resources, it should be via POST action.

Yes, the top level properties should be returned for proxy resource as well.

Incorrect resource requests and responses The response from any single resource GET, PUT, PATCH operation must be a JSON object with at least the following properties: id, name, type. Any RP-specific content that does not fit into the allowed top-level properties must be returned inside the properties property. id, name, type may be omitted in PUT or PATCH request bodies, but any RP-specific content must be present in the properties property in the request body. This applies to both Tracked and Proxy Only resource types.

Inconsistent or incomplete CRUD Any resource type MUST implement GET for collection and single resource regardless if it's a singleton or not.

Reply:
Thanks for explaining. I double-check with the owner of this new path. This is actually a requirement from Azure monitor because they use this get for auto scale. We are having the same model as vmss [VirtualMachineScaleSets_ListSkus](https://github.com/Azure/azure-rest-api-specs/blob/002ad52075e725708deabf842a736c98751b8566/specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/compute.json#L6028) which is a get and they return [VirtualMachineScaleSetListSkusResult](https://github.com/Azure/azure-rest-api-specs/blob/002ad52075e725708deabf842a736c98751b8566/specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/compute.json#L13885) which has an array of [VirtualMachineScaleSetSku](https://github.com/Azure/azure-rest-api-specs/blob/002ad52075e725708deabf842a736c98751b8566/specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/compute.json#L13865) and we have the same properties on our models NodeTypeListSkuResult and NodeTypeAvailableSku.